Fixes for running reward model inference using sglang

corbt opened the pull request:
<!-- Thank you for your contribution! We appreciate it. The following guidelines will help improve your pull request and facilitate feedback. If anything is unclear, don't hesitate to submit your pull request and ask the maintainers for assistance. -->

## Motivation

I'm trying to run a reward model (specifically a `LlamaForSequenceClassification`) with sglang. I ran into several bugs while doing so that this PR addresses.

## Modifications

1. Previously when serving a reward model, sglang attempted to pass `get_embedding=True` to the model's `forward()` method. That keyword doesn't exist, which caused calls to the model to fail. This change distinguishes embedding models from other non-generation models, which fixes the bug. I do this by adding a new `is_embedding_model` function which works similarly to the `is_generation_model` function.
2. Previously it was required to pass conversational input to reward models, which would then be serialized using the `apply_chat_template` function. However, sometimes it is convenient to pass in pre-templated strings instead, for example when using a model that doesn't have a chat template applied. This change allows reward models to accept strings in addition to chat formatted inputs. It accomplishes this by defining a new `_apply_chat_template` method on `TokenizerManager`
3. Skip copying the `lm_head` weights for models that don't have them defined. This previously caused a crash on startup for my `LlamaForSequenceClassification` model.

<!-- Describe the changes made in this PR. -->

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

corbt committed:
Reward model fixes

merrymercy commented on python/sglang/srt/model_executor/model_runner.py:
```diff
@@ -556,18 +559,18 @@ def forward_decode(self, forward_batch: ForwardBatch):
                 forward_batch,
                 get_embedding=True,
             )
+        else:
```
Why do you reverse this condition? Ideally, we want the common case to be the first branch. Could you put the generation case as the first branch?

corbt replied:
I reversed it because it felt like keeping the `if self.is_embedding` condition close to the embedding-specific functionality made the code a bit more readable. But I have switched it back now!

merrymercy reviewed (changes requested):
Thanks for the contribution! It looks good. I have one minor comment. Can you also add a unit test? 
https://github.com/sgl-project/sglang/tree/main/test

merrymercy commented:
@corbt Can you address the comments?

corbt committed:
put non-embedding models first

corbt commented:
Just addressed the comment!

I spent a bit of time looking at the test cases but didn't go deep on it. We are not deploying sglang in production immediately so I haven't had a chance to prioritize spending more time on this for now. Would it be better in that case for me to close the PR until I have a chance to invest more time in this?

merrymercy commented:
@corbt Thanks for the contribution. I cleaned them up and moved it to https://github.com/sgl-project/sglang/pull/1907
